### ISSUE TITLE
Failing to get detections with a GPU install on Windows 11

### ISSUE THREAD

**Issue by mambroset** [instanseg/instanseg#111]
Hi,
I tried a bunch of different conda environments to get the segmentation_minimal notebook to work with GPU but failed and am now at a loss what else to try:
Windows 11 Professional, i7, Nvidia T1000 (8GB), Python 3.11 for all envs

Creating the env with Python=3.11+ pip install instanseg-torch yields 
- "False" when trying python import torch, torch.cuda.is_available()
- "Requesting default device: cpu" and successful detections with cpu when trying the segmentation_minimal notebook:
<img width="1182" height="610" alt="Image" src="https://github.com/user-attachments/assets/5fe261a1-d21d-43f6-9363-324bb3d47f89" />

The same fresh env with either
conda install pytorch==2.5.1 torchvision torchaudio pytorch-cuda=12.4 -c pytorch -c nvidia
or pip install torch==2.6.0 torchvision==0.21.0 torchaudio==2.6.0 --index-url https://download.pytorch.org/whl/cu124
or pip install torch==2.7.0 torchvision==0.22.0 torchaudio==2.7.0 --index-url https://download.pytorch.org/whl/cu128
or other various combos of torch and cuda I've tried and forgotten, all in order to get "True" with torch.cuda.is_available()

all yield 
- "True" when trying python import torch, torch.cuda.is_available()
- "Requesting default device: cuda" in the notebook
- But no detections 
<img width="1182" height="610" alt="Image" src="https://github.com/user-attachments/assets/c4e49770-89e5-4307-9414-085f112e91ad" />

I've tried uninstalling torch and torchvision before pip install instanseg-torch,
I've tried to install the correct torch/cuda combo before pip install instanseg-torch, 
I've tried installing cupy-cuda12x as well
I've also tried cloning the repo and doing the install with pip install -e .

Nothing changed anything
The documentation seems a bit outdated for this GPU part, and it's unclear that it should be optional and only if pip install instanseg-torch does not work out of the box (randomly found out while digging in old issues :))

Am I missing some crucial point?
Thanks in advance for your help!
Melow

**Comment by alanocallaghan** [comment]
I can confirm that the default install of `instanseg[full]` doesn't include CUDA, but after installing cu124 2.6.0 with
```
pip install torch==2.6.0 torchvision==0.21.0 torchaudio==2.6.0 --index-url https://download.pytorch.org/whl/cu124
```

I get valid detections:

<img width="1322" height="540" alt="Image" src="https://github.com/user-attachments/assets/2a754aff-6764-4e93-aa9c-c46d34d12d46" />

I can't currently reproduce the linked issue with the extension either, but I'm still working on that.

**Comment by mambroset** [comment]
So, I installed miniforge in WSL and even though I managed to get cuda without doing anything else than pip install instanseg-torch[full], I also get no detections in the notebook with that environment

I'm quite new at all of this, is there another way I'm supposed to use WSL? Thanks!

**Comment by alanocallaghan** [comment]
Interesting, in that case it seems unlikely that windows is related.

Can you share the code you are using, and reproduce the issue on [an example image](https://openslide.cs.cmu.edu/download/openslide-testdata/)?

**Comment by mambroset** [comment]
I am using the cloned repository notebook, with the provided examples (e.g. HE_example.tif, LuCa1.tif)
I either use the notebook from Pycharm or from jupyter lab

The very odd thing is that I just noticed the eval_medium_image() method works, while the eval() and eval_small_image() don't!
``` Python
image_array, pixel_size = instanseg_brightfield.read_image("../instanseg/examples/HE_example.tif")

labeled_output, image_tensor  = instanseg_brightfield.eval_medium_image(image_array, pixel_size)

display = instanseg_brightfield.display(image_array, labeled_output)

show_images(image_array,display, colorbar=False, titles = ["Original Image", "Image with segmentation"])
```

<img width="1182" height="610" alt="Image" src="https://github.com/user-attachments/assets/0e8319d6-12ce-43b1-93ac-0ea486a36569" />
I do get the following warning in the output:

.../instanseg/instanseg/utils/pytorch_utils.py:204: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at /pytorch/aten/src/ATen/SparseCsrTensorImpl.cpp:53.)
  intersection = torch.sparse.mm(onehot1, onehot2.T).to_dense()


But

``` Python
labeled_output = instanseg_brightfield.eval(image = "../instanseg/examples/HE_example.tif",
                                            save_output = True,
                                            save_overlay=True)
show_images(labeled_output)
```

<img width="855" height="781" alt="Image" src="https://github.com/user-attachments/assets/74b52f4a-7626-4749-ad05-6cf2bb9b7cac" />

and

``` Python
image_array, pixel_size = instanseg_brightfield.read_image("../instanseg/examples/HE_example.tif")

labeled_output, image_tensor  = instanseg_brightfield.eval_small_image(image_array, pixel_size)

display = instanseg_brightfield.display(image_array, labeled_output)

show_images(image_array,display, colorbar=False, titles = ["Original Image", "Image with segmentation"])
```

<img width="1182" height="610" alt="Image" src="https://github.com/user-attachments/assets/eba52406-dc86-4ee0-87f7-7dfaa33035ea" />

**Comment by mambroset** [comment]
Ran
``` Python
image_array, pixel_size = instanseg_brightfield.read_image("../instanseg/examples/HE_example.tif")

labeled_output, image_tensor  = instanseg_brightfield.eval_small_image(image_array, pixel_size)

show_images(labeled_output, image_tensor)

labeled_output, image_tensor  = instanseg_brightfield.eval_small_image(image_tensor[:,:,:512,:512], pixel_size)

show_images(labeled_output, image_tensor)
```
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers). Got range [-0.13609467..1.0697675].

<img width="1182" height="588" alt="Image" src="https://github.com/user-attachments/assets/e05ee737-fad8-493e-9677-161c82f70c30" />

Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers). Got range [-0.10588235..1.051402].
<img width="1182" height="548" alt="Image" src="https://github.com/user-attachments/assets/34ab82bf-06f6-445d-88b3-388fb0228fb5" />

**Comment by ThibautGoldsborough** [comment]
Hmm, I wonder if it is a CUDA memory error in that case. Strange that you don't get an explicit error message. You should stick to use eval_medium_image in the meantime.

 I think it will be hard for us to fix without having access to your specific gpu.

**Comment by mambroset** [comment]
Something maybe similar to this https://github.com/napari/napari/pull/8122? My GPU was among the annoying ones for that particular problem :)
